llvm/lib/Target/AIE/AIEBaseInstrInfo.cpp

```diff
@@ -184,18 +195,18 @@ bool AIEBaseInstrInfo::analyzeBranch(MachineBasicBlock &MBB,
   // Handle a conditional branch followed by an unconditional branch.
   if (NumTerminators == 2 && std::prev(I)->getDesc().isConditionalBranch() &&
       I->getDesc().isUnconditionalBranch()) {
-    // PseudoLoopEnd
-    if (isHardwareLoopEnd(std::prev(I)->getOpcode())) {
-      TBB = std::prev(I)->getOperand(0).getMBB();
-      Cond.push_back(MachineOperand::CreateImm(std::prev(I)->getOpcode()));
-      Cond.push_back(std::prev(I)->getOperand(0));
-      Cond.push_back(std::prev(I)->getOperand(1));
-    } else
+    auto Prev = std::prev(I);
+    if (isHardwareLoopEnd(Prev->getOpcode())) {
+      // PseudoLoopEnd
+      TBB = Prev->getOperand(0).getMBB();
+      Cond.push_back(MachineOperand::CreateImm(Prev->getOpcode()));
+      Cond.push_back(Prev->getOperand(1));
+    } else {
       // JNZD
-      if (!(std::prev(I)->getOperand(1).isMBB()))
+      if (!Prev->getOperand(1).isMBB())
         return Unhandled;
-      else
-        parseCondBranch(*std::prev(I), TBB, Cond);
+      parseCondBranch(*Prev, TBB, Cond);
+    }
```

Comment:
Nit: maybe let `parseCondBranch` determine whether it can actually parse the branch? E.g.
```
else if (!parseCondBranch(*Prev, TBB, Cond)) {
  return Unhandled;
}
```